### PR TITLE
Do not crash on empty build/host/run sections

### PR DESCRIPTION
**Pull request by saraedum**
fixes this crash: https://github.com/conda-forge/nauty-feedstock/pull/7#issuecomment-540670699

Fixes https://github.com/conda-forge/conda-smithy/issues/1153

### CODE REVIEW

**Comment by jakirkham** [comment]
Can you please add a test for this @saraedum?

**Comment by saraedum** [comment]
I feared that this would happen ;)

File: tests/test_lint_recipe.py [hunk 1]
```diff
@@ -301,6 +301,7 @@ def assert_noarch_selector(meta_string, is_good=False):
                               requirements:
                                 build:
                                   - python
+                                host: # empty sections are allowed and ignored
```
Comment (jakirkham) [inline tests/test_lint_recipe.py]: Should we go ahead and add `run` too?

**Comment by jakirkham** [comment]
@isuruf @scopatz, can you please also take a quick look? 🙂

**Pull request merged**